dlt/extract/incremental/transform.py

```diff
@@ -206,24 +232,19 @@ def __call__(
         cursor_path = str(self.cursor_path)
         # The new max/min value
         try:
-            row_value = compute(tbl[cursor_path]).as_py()
+            orig_row_value = compute(tbl[cursor_path])
+            row_value = orig_row_value.as_py()
+            # dates are not represented as datetimes but I see connector-x represents
+            # datetimes as dates and keeping the exact time inside. probably a bug
+            # but can be corrected this way
+            if isinstance(row_value, date) and not isinstance(row_value, datetime):
+                row_value = pendulum.from_timestamp(orig_row_value.cast(pa.int64()).as_py() / 1000)
```

Comment:
Interesting. Should we have a test for this?  
Not common probably, but initial and end values should be converted the same way if they're dates so they're comparable.

Reply:
this is a bug in connector x. when I use arrow2 type with binary protocol, the types are correct. if the end and start values are dates, they will compare with datetimes just fine. it was working in my (manual) tests